src/Acrobot.jl

```diff
@@ -83,27 +89,27 @@ function step!(env::AcrobotEnv, action)
     # only care about final timestep of integration returned by integrator
     ns = ns[end, :]
     ns = ns[1:4]  # omit action
-    ns[1] = wrap(ns[1], -pi, pi)
-    ns[2] = wrap(ns[2], -pi, pi)
-    ns[3] = bound(ns[3], -env.MAX_VEL_1, env.MAX_VEL_2)
-    ns[4] = bound(ns[4], -env.MAX_VEL_1, env.MAX_VEL_2)
+    ns.data[1] = wrap(ns[1], -pi, pi).data
+    ns.data[2] = wrap(ns[2], -pi, pi).data
+    ns.data[3] = bound(ns[3], -env.MAX_VEL_1, env.MAX_VEL_2).data
+    ns.data[4] = bound(ns[4], -env.MAX_VEL_1, env.MAX_VEL_2).data
     env.state = ns
     terminal = _terminal(env)
-    reward = !terminal ? -1f0 : 0f0
+    reward = !all(terminal) ? -1f0 : 0f0
 
-    return (_get_ob(env), reward, terminal, Dict())
+    return _get_ob(env), reward, terminal, Dict()
 end
 
 
 function _get_ob(env::AcrobotEnv)
     s = env.state
-    return [cos(s[1]), sin.(s[1]), cos(s[2]), sin(s[2]), s[3], s[4]]
+    return vcat(cos.(s[1:1]), sin.(s[1:1]), cos.(s[2:2]), sin.(s[2:2]), s[3:3], s[4:4])
 end
 
 
 function _terminal(env::AcrobotEnv)
     s = env.state
-    return (cos(s[1])-cos(s[2]+s[1])) .> 1f0
+    return (cos.(s[1:1])-cos.(s[2:2]+s[1:1])) .> 1f0
 end
 
 
```

Comment:
The following function calculates the gradients for Runge Kutta. Can we use Flux here?

Reply:
If we have Runge Kutta defined somewhere, then yes.

Reply:
`OrdinaryDiffEq.jl` has RK4 and other solvers defined which Flux can differentiate through. (You can find an example usage [here](https://github.com/Roboneet/Trebuchet.jl/blob/master/src/simulate.jl#L22L53) and [here](https://github.com/FluxML/model-zoo/blob/master/games/differentiable-programming/trebuchet/DiffRL.jl#L12))